src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectApplication.groovy

```diff
@@ -0,0 +1,114 @@
+package life.qbic.datamodel.dtos.projectmanagement
+
+import life.qbic.datamodel.dtos.business.Customer
+import life.qbic.datamodel.dtos.business.OfferId
+import life.qbic.datamodel.dtos.business.ProjectManager
+
+/**
+ * Information about a desired project registration in QBiC's project management system.
+ *
+ * Instances of this class describe a request to register a new project at QBiC.
+ *
+ * Beside the mandatory fields required to be passed to the constructor, there are two optional
+ * fields:
+ *   1: projectSpace
+ *   2: projectCode
+ * If any of these are not defined, then either of it will be created randomly.
+ *
+ * @since 2.3.0
+ */
+class ProjectApplication {
+
+    /**
+     * The associated offer
+     */
+    final OfferId linkedOffer
+
+    /**
+     * A short and precise project title
+     */
+    final String projectTitle
+
+    /**
+     * A descriptive project objective
+     */
+    final String projectObjective
+
+    /**
+     * A description about the experimental design
+     */
+    final String experimentalDesign
+
+    /**
+     * The associated project manager
+     */
+    final ProjectManager projectManager
+
+    /**
+     * The associated customer
+     */
+    final Customer customer
+
+    /**
+     * The requested project space the project shall be associated with (optional)
+     */
+    final Optional<ProjectSpace> projectSpace
+
+    /**
+     * The desired project code (optional)
+     */
+    final Optional<ProjectCode> projectCode
```

Comment:
I think they should not be optional

Reply:
yes, they can be :) if the code is not set, one will be provided then.